web_poet/testing/pytest.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+import json
+import operator
+from pathlib import Path
+from typing import Any, Iterable, List, Optional, Set, Union
+
+import pytest
+from freezegun import freeze_time
+
+from web_poet import ItemPage
+from web_poet.serialization import SerializedDataFileStorage, deserialize, load_class
+from web_poet.testing import INPUT_DIR_NAME, OUTPUT_FILE_NAME
+from web_poet.testing.utils import META_FILE_NAME
+from web_poet.utils import ensure_awaitable
+
+
+class WebPoetFile(pytest.File):
+    """Represents a directory containing test case subdirectories for one Page Object."""
+
+    @staticmethod
+    def sorted(items: List["WebPoetItem"]) -> List["WebPoetItem"]:
+        """Sort the item list by the item name."""
+        return sorted(items, key=operator.attrgetter("name"))
+
+    def collect(self) -> Iterable[Union[pytest.Item, pytest.Collector]]:  # noqa: D102
+        result: List[WebPoetItem] = []
+        for entry in self.path.iterdir():
+            if entry.is_dir():
+                item: WebPoetItem = WebPoetItem.from_parent(
+                    self,
+                    name=entry.name,
+                    path=entry,
+                )
+                if item.input_path.is_dir() and item.output_path.is_file():
+                    result.append(item)
+        return self.sorted(result)
+
+
+class WebPoetItem(pytest.Item):
+    """Represents a directory containing one test case."""
```

Comment:
It seems we use 3 different terms for the same concept:

* Item (I guess it comes from pytest)
* test case (not sure a name is good; in unittest TestCase is a collection of tests
* fixture (save_fixture, etc.)

None of this fits the definitions in https://docs.python.org/3/library/unittest.html. I think pytest uses "test" term for this.

Not sure what's the best way, but it'd be good to clean it up.